tests/test_model.py

```diff
@@ -89,3 +91,73 @@ def test_test_execution_output_dictionary_with_test_execution_id(testcase, date_
                 }
             ]
         }
+
+
+def test_test_execution_full_model(testcase, date_time_now):
+    with patch('datetime.datetime') as dt_mock:
+        dt_mock.now.return_value = date_time_now
+        te = _TestExecution(
+            test_plan_key='Jira-10',
+            test_execution_key='JIRA-20',
+            test_environments=["My local laptop"],
+            fix_version="1.0",
+            summary="My Test Suite",
+            description="Im doing stuff"
+        )
+        te.tests = [testcase]
+        assert te.as_dict() == {
+            'testExecutionKey': 'JIRA-20',
+            'info': {
+                'finishDate': '2021-04-23T16:30:02+0000',
+                'startDate': '2021-04-23T16:30:02+0000',
+                'testPlanKey': 'Jira-10',
+                'version': '1.0',
+                'testEnvironments': [
+                    'My local laptop'
+                ],
+                'summary': 'My Test Suite',
+                'description': 'Im doing stuff'
+            },
+            'tests': [
+                {
+                    'comment': 'Test',
+                    'status': 'PASS',
+                    'testKey': 'JIRA-1'
+                }
+            ]
+        }
+
+
+def test_test_execution_environ_model(testcase, date_time_now):
+
+    os.environ[constant.ENV_TEST_EXECUTION_FIX_VERSION] = "1.1"
```

Comment:
This may affect other tests which are executed after this one. It will be safer to patch os.environ, e.g.:
```
@mock.patch.dict(os.environ, {
    constant.ENV_TEST_EXECUTION_FIX_VERSION: "1.1",
    constant.ENV_TEST_EXECUTION_TEST_ENVIRONMENTS: "MyLocalLaptop And TheLiveSystem",
})
def test_test_execution_environ_model(testcase, date_time_now):
```

Reply:
Thanks. I didn't know about that feature. I will do it on monday. Have a nice weekend.

Reply:
Here you can find examples https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pytest/